CHANGELOG.md

```diff
@@ -5,202 +5,140 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [0.20.3] - 2024-04-21
+## Unreleased
+### Added
+- GHA: use `cargo-udeps` to prevent the possibility of using unused dependencies
+### Changed
+- GHA: use `cargo-binstall` to speed up CI jobs
 
+## [0.20.3] - 2024-04-21
 ### Fixed
-
--   Missing compacting service dependency
+- Missing compacting service dependency
```

Comment:
One dev experience improvement we should make is to use same auto-formatters regardless of IDE - e.g. configuring something like `prettier` to handle markdown files / toml / yaml / etc. Mb let's cut a ticket for this in our roadmap.

But for now, please revert the auto-formatting changes.

Reply:
> But for now, please revert the auto-formatting changes.

Inversely, my commit is a revert of the changes we accidentally merged earlier